common/src/main/java/com/microsoft/identity/common/internal/request/IAuthenticationCallback.java

```diff
@@ -0,0 +1,58 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.common.internal.request;
+
+import com.microsoft.identity.common.exception.BaseException;
+import com.microsoft.identity.common.internal.result.IBaseAuthenticationResult;
+
+/**
+ * Callback passed with token acquisition. {@link IBaseAuthenticationResult} or {@link Exception} will be returned back via callback.
+ */
+public interface IAuthenticationCallback {
```

Comment:
Same as how the the internal `IAuthenticationCallback` is separate from `AuthenticationCallback`, the `IBaseAuthenticationResult` and `IAuthenticationResult` should be separate.

Reply:
The name of this class seems like a possible source of confusion.... since we have `IAuthenticationCallback` and `AuthenticationCallback`.

The public API should probably not change at this point so maybe something in this class' name to signal that it's internal?

Reply:
Issue opened here: https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/478